### PR TITLE
refactor(common): unify order-related types

### LINKED ISSUE #7591: refactor: unify types related to row sorting
Types representing order direction:

- `risingwave_frontend::optimizer::property::order::Direction`
- `risingwave_common::util::sort_util::OrderType`

Types representing order-by field:

- `risingwave_frontend::optimizer::property::order::FieldOrder`
- `risingwave_frontend::optimizer::plan_node::generic::agg::PlanAggOrderByField`
- `risingwave_common::util::sort_util::OrderPair`

Protobuf types:

- `expr::OrderByField`
- `plan_common::ColumnOrder`

### PR DESCRIPTION
**Pull request by richardchien**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- `Direction` becomes a field of `OrderType`
- `OrderPair`, `FieldOrder` and `PlanAggOrderByField` are replaced by `ColumnOrder`, corresponding to `ColumnOrder` in proto
- Binding support for `NULLS FIRST | LAST` are temporarily disabled, will be supported in #4743, by enhancing `OrderType` struct

Main changes happen in files listed below, others are just renaming and necessary adaptation for new types:

- `src/common/src/util/sort_util.rs`
- `src/frontend/src/binder/expr/order_by.rs`
- `src/frontend/src/expr/order_by_expr.rs`
- `src/frontend/src/optimizer/plan_node/generic/agg.rs`
- `src/frontend/src/optimizer/property/order.rs`

This PR will fix #7591.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/common/src/util/sort_util.rs [hunk 1]
```diff
@@ -13,78 +13,179 @@
...
+        }
+    }
+}
+
+impl OrderType {
+    pub const fn new(direction: Direction) -> Self {
+        Self { direction }
+    }
+
+    // Create an ascending order type, with other options set to default.
```
Comment (BugenZhao) [inline src/common/src/util/sort_util.rs:91]:
```suggestion
    /// Create an ascending order type, with other options set to default.
```

Let make these comments rustdoc.

File: src/common/src/util/sort_util.rs [hunk 2]
```diff
@@ -13,78 +13,179 @@
...
+    }
+
+    /// Create an ascending order type, with other options set to default.
+    pub const fn ascending() -> Self {
+        Self {
+            direction: Direction::Ascending,
+        }
+    }
+
+    // Create an descending order type, with other options set to default.
```
Comment (BugenZhao) [inline src/common/src/util/sort_util.rs]: Update all comments? 🥵

**Review by BugenZhao — COMMENTED** [review]
Rest LGTM

File: src/common/src/util/sort_util.rs [hunk 3]
```diff
@@ -13,78 +13,179 @@
...
+
+#[allow(dead_code)]
+#[derive(PartialEq, Eq, Hash, Copy, Clone, Debug, Display, Default)]
+pub enum NullsAre {
+    #[default]
+    #[display("LARGEST")]
+    Largest,
+    #[display("SMALLEST")]
+    Smallest,
+}
```
[the remark this answers is not in the document]
Comment (st1page) [reply src/common/src/util/sort_util.rs:63]: If we need compactable with PG‘s default behavior, I prefer to implement it in the `OrderType::with_direction()` I think it depends on the memcomparable's implementation and we can determine it later.

**Review by st1page — APPROVED** [review]
generally LGTM

File: src/common/src/util/sort_util.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (richardchien) [reply src/common/src/util/sort_util.rs:63]: Found that in memcomparable encoding, our current behavior is `NULLS LAST`, while in `compare_values(lhs: T, rhs: T, order_type)`, the behavior is `NULLS LARGEST`🥵. Anyway, NULLs first/last is indeed better. Will use this notation later when enabling `NULLS FIRST | LAST` support.

**Pull request merged**